module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,46 @@
+#include "compression.hpp"
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap) {
+    
+    CompressedBitmap compressed;
+    uint8_t color;
+    uint8_t count;
+
+    for (size_t i = 0; i < height; i++) {
+
+        color = bitmap[i][0];
+        count = 1;
+
+        for (size_t j = 1; j < width; j++) {
+            if (color == bitmap[i][j]) {
+                count++;
+            } else {
+                compressed.emplace_back(std::make_pair(color, count));
+                color = bitmap[i][j];
+                count = 1;
+            }
+        }
+        compressed.emplace_back(std::make_pair(color, count));
+    }
+
+    return compressed;
+}
```

Comment:
Good usage of aliases, in my opinion too much spaces, and you can rearange loop to use emplace_back just once. Another things is that using std::make_pair is not necessary you can make pair just by using this notation {first, second}

Reply:
Agree with @jakjan95 

Please try use `empalce_back` once. And also try to use iterators it will be simpler ;)

Reply:
done